tests/manage/cluster/test_must_gather.py

```diff
@@ -0,0 +1,182 @@
+import os
+import logging
+import time
+
+import pytest
+
+from ocs_ci.framework.testlib import ManageTest, tier1, bugzilla
+from ocs_ci.ocs import openshift_ops, ocp
+from ocs_ci.ocs.utils import collect_ocs_logs
+from ocs_ci.utility.utils import ocsci_log_path
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+@pytest.mark.polarion_id("OCS-1583")
+@bugzilla('1766646')
+class TestMustGather(ManageTest):
+
+    @pytest.fixture(autouse=True)
+    def init_ocp(self):
+        """
+        init OCP() object
+        """
+
+        self.ocs = openshift_ops.OCP()
+        self.ocp_obj = ocp.OCP()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+
+        def finalizer():
+            must_gather_pods = self.ocs.get_pods(label_selector='app=must-gather')
+            logger.info(f"must_gather_pods: {must_gather_pods} ")
+            if must_gather_pods:
+                for pod_to_del in must_gather_pods:
+                    self.ocp_obj.wait_for_delete(resource_name=pod_to_del)
+                    logger.info(f"deleted pods: {pod_to_del}")
```

Comment:
since you're not actively deleting the must gather pod, this log message and the pod variable name can be confusing. I'd rephrase the pod var to 'must_gather_pod' and the log msg to "Pod {must_gather_pod} got deleted successfully"